src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -218,13 +215,11 @@ static SourceGeneratorExecutionVersionMap FilterToProjectCone(SourceGeneratorExe
                     oldProjectIdToStateChecksums.Add(projectId, oldProjectStateChecksums);
                 }
 
-                // sync over the *info* about all the added/changed projects.  We'll want the info so we can determine
-                // what actually changed.
                 using var _5 = PooledHashSet<Checksum>.GetInstance(out var newChecksumsToSync);
                 newChecksumsToSync.AddRange(newProjectIdToChecksum.Values);
 
                 var newProjectStateChecksums = await _assetProvider.GetAssetsAsync<ProjectStateChecksums>(
-                    assetHint: AssetHint.None, newChecksumsToSync, cancellationToken).ConfigureAwait(false);
+                    assetHint: AssetHint.SolutionAndTopLevelProjectsOnly, newChecksumsToSync, cancellationToken).ConfigureAwait(false);
```

Comment:
the other place where we pass SolutionAndTopLevelProjectsOnly since we're bulk syncing just the project data over.  as per above, i'd like to change this to potentially be N project calls.  but it needs to be measured.